model/task_lifecycle.go

```diff
@@ -1781,22 +1781,20 @@ func checkResetDisplayTask(t *task.Task) error {
 		}
 	}
 	details := &t.Details
-	if details == nil && !t.IsFinished() {
+	// assign task end details to indicate system failure if we receive no valid details
+	if taskEndDetailsIsEmpty(details) && !t.IsFinished() {
 		details = &apimodels.TaskEndDetail{
 			Type:   evergreen.CommandTypeSystem,
 			Status: evergreen.TaskFailed,
 		}
 	}
-	grip.DebugWhen(details.Status == "", message.Fields{
-		"message":   "resetting display task",
-		"task_id":   t.Id,
-		"execution": t.Execution,
-		"project":   t.Project,
-		"details":   details,
-	})
 	return errors.Wrap(TryResetTask(t.Id, evergreen.User, evergreen.User, details), "resetting display task")
 }
 
+func taskEndDetailsIsEmpty(details *apimodels.TaskEndDetail) bool {
```

Comment:
Could it be reasonable to make this a method on TaskEndDetail?

Reply:
Could possibly also use here https://github.com/evergreen-ci/evergreen/blob/d12940c1609c7e47d27c9a1fddf7ee2972a743a2/model/task/task.go#L1840